### PR TITLE
Issue-31 Export Logs

### PR DESCRIPTION
**Pull request by anyercastillo**
Re-uses the existing module for export logs

### Pending
1- Discuss with the team how to deploy the changes to maven.
2- Remove reference on `app` to local project and reference the maven library.

### Video
https://github.com/getditto/DittoAndroidTools/assets/5747647/dcd22544-86ba-4d3d-a844-51b9818e25d4

### CODE REVIEW

**Review by phatblat — APPROVED** [review]

**Review by okdistribute — CHANGES_REQUESTED** [review]
Ah this is confusing but for https://github.com/getditto/DittoAndroidTools/issues/31 we actually want to export the data directory not the log files. The ditto directory contains important information about the state of the database which can help technical support diagnose issues in the case it is bloated, corrupted, or slow. 

See https://github.com/getditto/DittoSwiftTools/blob/main/Sources/DittoExportData/ExportData.swift#L41 for the Swift example

File: DittoDiskUsage/build.gradle [hunk 1]
```diff
@@ -19,11 +19,13 @@ dependencies {
 
     debugImplementation libs.androidx.compose.ui.uiTooling
 
+    implementation libs.androidx.compose.material.iconsExtended
```
[the remark this answers is not in the document]
Comment (anyercastillo) [reply DittoDiskUsage/build.gradle]: Good Catch!
Thank you so much.

**Comment by anyercastillo** [comment]
The new refactoring takes care of:
- reuse export logic across modules
- export ditto directory
- refactor `DittoExportLogs` and `DittoDiskUsage` to use the common export logic
- remove dependency on Extended Icons and only uses the FolderZip resource
- updates the Export Dialog to make it reusable and also to include a loading indicator for long time operations

https://github.com/getditto/DittoAndroidTools/assets/5747647/e31b62b3-d898-4140-96b7-30f28ab42348

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 2]
```diff
@@ -1,52 +1,23 @@
...
-        dismissButton = {
-            Button(onClick = onDismiss) {
-                Text("Cancel")
-            }
-        }
+    ExportDialog(
+        title = "Export Logs",
+        text = "Do you want to export logs?",
+        confirmText = "Export",
+        cancelText = "Cancel",
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:13]: Should these be in a `strings.xml`?

File: DittoExporter/src/test/java/live/ditto/dittoexportlogs/ExampleUnitTest.kt
Comment (rajramsaroop) [inline DittoExporter/src/test/java/live/ditto/dittoexportlogs/ExampleUnitTest.kt:1]: Might as well delete this file!

File: DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.core.net.toFile
+import kotlinx.coroutines.launch
+import java.io.File
+import java.io.FileInputStream
+
+/**
+ * A customized AlertDialog to export the given [file]
```
Comment (rajramsaroop) [inline DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt:29]: Change to `[File]` so it references the `java.io.File`

File: DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.core.net.toFile
```
Comment (rajramsaroop) [inline DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt:23]: unused import

File: DittoExporter/src/test/java/live/ditto/dittoexportlogs/ExampleUnitTest.kt
Comment (anyercastillo) [reply DittoExporter/src/test/java/live/ditto/dittoexportlogs/ExampleUnitTest.kt:1]: It is there for all the other modules. 
I guess it is a reason for it.

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt (hunk 2 above)
Comment (anyercastillo) [reply DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:13]: Unrelated with the PR, but happy to take care of it.

File: DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt (hunk 3 above)
Comment (anyercastillo) [reply DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt:29]: It references the property `file` not the data type

Comment (anyercastillo) [reply DittoExporter/src/main/java/live/ditto/exporter/ExportDialog.kt:29]: oh the issue is the property name was change from `file` to `fileProvider` and for whatever reason AS did not change it.

**Review by rajramsaroop — APPROVED** [review]

**Review by okdistribute — APPROVED** [review]

**Pull request merged**